internal/rpc/rpc_handler.go

```diff
@@ -0,0 +1,172 @@
+package rpc
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	hadoop "github.com/colinmarc/hdfs/v2/internal/protocol/hadoop_common"
+	"github.com/golang/protobuf/proto"
+	"gopkg.in/jcmturner/gokrb5.v5/crypto"
+	"gopkg.in/jcmturner/gokrb5.v5/gssapi"
+	"gopkg.in/jcmturner/gokrb5.v5/iana/keyusage"
+	krbtypes "gopkg.in/jcmturner/gokrb5.v5/types"
+)
+
+// To check interface validity
+var rpcWriter RpcWriter = &BasicRpcWriter{}
+var rpcReader RpcReader = &BasicRpcReader{}
+var saslRpcReader RpcReader = &SaslRpcReader{}
+
+// RpcWriter is an interface for sending RPC payload
+type RpcWriter interface {
```

Comment:
I think this is basically, right, but can you call it 'Transport'? There's a good chance the Transport should track the request sequence number, too.

Reply:
Could you tell me what is supposed to call `Transport` in detail?
I suppose your comment above is more than just renaming the `RpcWriter` interface.